src/hotspot/share/classfile/classFileParser.cpp

```diff
@@ -5916,7 +5931,9 @@ void ClassFileParser::verify_legal_method_name(const Symbol* name, TRAPS) const
 
   if (length > 0) {
     if (bytes[0] == JVM_SIGNATURE_SPECIAL) {
-      if (name == vmSymbols::object_initializer_name() || name == vmSymbols::class_initializer_name()) {
+      if (name == vmSymbols::object_initializer_name() ||
+          name == vmSymbols::class_initializer_name() ||
+          name == vmSymbols::inline_factory_name()) {
```

Comment:
Shouldn't we check that the class file supports inline types before accepting a static factory name?